src/praisonai/tests/unit/sandbox/test_ssh.py

```
@@ -0,0 +1,249 @@
+        """Test writing file to remote server."""
+        mock_connection = AsyncMock()
+        mock_sftp = AsyncMock()
+        mock_file = AsyncMock()
+        mock_sftp.open.return_value.__aenter__.return_value = mock_file
+        mock_connection.start_sftp_client.return_value.__aenter__.return_value = mock_sftp
+        
+        sandbox = SSHSandbox(host="test.example.com")
+        sandbox._connection = mock_connection
+        sandbox._is_running = True
+        
+        success = await sandbox.write_file("/tmp/test.py", "print('Hello')")
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/test.py"

(S108)